### ISSUE TITLE
Support for `lazy` imports (PEP 810, Python 3.15+)

### ISSUE THREAD

**Issue by cjw296** [astral-sh/ty#2968]
Python 3.15 introduces [PEP 810 - Explicit lazy imports](https://peps.python.org/pep-0810/) via the new `lazy` soft keyword:

```python
lazy import json
lazy from pathlib import Path
```

Currently `ty` reports these as syntax errors. Support is needed to:

- [ ] Update the parser to recognise `lazy import` and `lazy from ... import` statements
- [ ] Treat lazy imports the same as regular imports for type checking purposes (as noted in [PEP 810 § Typing and tools](https://peps.python.org/pep-0810/#typing-and-tools))

Related: astral-sh/ruff#21305 (ruff tracking issue for the same feature)

**Comment by carljm** [comment]
I suspect that this will be a single PR to the parser to fix both ruff and ty, since ruff and ty share the parser, and ty shouldn't need to do anything beyond accepting this in the parser. But doesn't hurt to have separate issues open. Thanks for reporting!

**Comment by charliermarsh** [comment]
I think this is "done" in a sense? In the next release, we support parsing these.
